Source/WebCore/dom/DocumentMarker.h

```diff
@@ -126,6 +127,18 @@ class DocumentMarker {
     void setEndOffset(unsigned offset) { m_range.end = offset; }
     void shiftOffsets(int delta);
 
+    DocumentMarker(DocumentMarker&& other)
+        : m_type(other.m_type)
+        , m_range(other.m_range)
+        , m_data(WTFMove(other.m_data))
+    { }
+
+    DocumentMarker(const DocumentMarker& other)
```

Comment:
Does `DocumentMarker(const DocumentMarker& other) = default;` not work in this case?

Reply:
No, because the new base class `CanMakeWeakPtr` doesn't support copying.

Reply:
I think we should strive to avoid unnecessary lists of data members that can fall out of date. Maybe we should start using structures in cases like this so we can take advantage of copy constructors and == operators without listing data members. New constraints might lead us to new coding idioms.

Reply:
Ok, I'll add the copy constructor to CanMakeWeakPtr to avoid this.